src/main/java/mate/academy/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.dao;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.db.Storage;
+import mate.academy.lib.Dao;
+import mate.academy.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(m -> m.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        get(manufacturer.getId()).get().setCountry(manufacturer.getCountry());
+        get(manufacturer.getId()).get().setName(manufacturer.getName());
```

Comment:
let's iterate list -> find object with the same id and than use set. Current approach won't work correctly because ids and indexes may not be the same, when we delete object from list for example. Plus, when you've found needed object set the whole manufacturer instead of setting each field, this way if you want to add another field to manufacturer you won't need to rewrite this method